src/nvim/api/win_config.c

```diff
@@ -639,6 +656,51 @@ void nvim_win_set_config(Window window, Dict(win_config) *config, Error *err)
       win_setheight_win(fconfig.height, win);
     }
   } else {
+    if (win->w_floating && parent && parent_tp != win_tp) {
+      if (!can_move(win, win_tp != parent_tp, err)) {
+        return;
+      }
+
+      if (win == curwin) {
+        win_goto(win_float_find_altwin(win, NULL));
+        if (curwin == win) {
+          api_set_error(err, kErrorTypeException, "Failed to switch away from window %d",
+                        win->handle);
+          return;
+        }
+      }
+      win_remove(win, win_tp == curtab ? NULL : win_tp);
+
+      // Find insertion point in target tabpage
+      win_T *target_after;
+      if (parent_tp == curtab) {
+        target_after = lastwin_nofloating();
+      } else {
+        target_after = parent_tp->tp_lastwin;
+        while (target_after && target_after->w_floating) {
+          target_after = target_after->w_prev;
+        }
+        if (target_after == NULL) {
+          target_after = parent_tp->tp_lastwin;
+        }
+      }
+
+      // Insert into target tabpage
+      win_append(target_after, win, parent_tp == curtab ? NULL : parent_tp);
+      // Ensure target tabpage has valid curwin
+      if (parent_tp != curtab && parent_tp->tp_curwin == NULL) {
+        parent_tp->tp_curwin = parent_tp->tp_firstwin;
+      }
```

Comment:
Not sure what this is checking. Why would `parent_tp->tp_curwin` be NULL here?

You _should_ however check that `win_tp->tp_curwin` is set to something valid if it was previously equal to `win`, like is done for splits.